### PR TITLE
Separate lifecycle callbacks for activity + process state

### PR DESCRIPTION
**Pull request by fractalwrench**
## Goal

Separates the lifecycle callbacks for the activity + process state. Currently these are implemented together in the same class   (`ActivityService`) but these are distinct responsibilities so we should separate them.

I have refactored the implementation so that `ProcessStateService` is responsible for tracking the process foreground/background state, and that `ActivityTracker` is responsible for tracking activity lifecycle callbacks.

## Testing

Updated existing test coverage.

### CODE REVIEW

File: embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/EmbraceImpl.java [hunk 1]
```diff
@@ -1518,8 +1526,13 @@
...
 
     @Nullable
-    ActivityService getActivityService() {
-        return activityService;
+    ProcessStateService getActivityService() {
+        return processStateService;
+    }
+
+    @Nullable
+    ActivityTracker getActivityLifecycleTracker() {
```
Comment (bidetofevil) [inline embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/EmbraceImpl.java:1534]: Who calls this? If we don't need this I'd rather not have it

Comment (fractalwrench) [reply embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/EmbraceImpl.java:1534]: `EmbraceAutomaticVerification` calls it (and other services) unfortunately

Comment (bidetofevil) [reply embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/EmbraceImpl.java:1534]: Ack....

**Review by bidetofevil — APPROVED** [review]
Looks good to me

**Pull request merged**